src/components/drop-down/drop-down.tsx

```diff
@@ -10,12 +11,34 @@ export interface DropDownInputProps {
 }
 
 export const DropDownInput: React.SFC<DropDownInputProps> = SBStateless(props => {
-  return (
-      <div data-automation-id="DROP_DOWN_INPUT" onClick={props.onClick} className="drop-down-input">
-          <span className="label">{props.selectedItem ? props.selectedItem.label : 'Default Text'}</span>
-          <CaretDown className="caret" />
-      </div>
-  );
+    return (
+        <div data-automation-id="DROP_DOWN_INPUT" onClick={props.onClick} className="drop-down-input">
+            <span className="label">{props.selectedItem ? props.selectedItem.label : 'Default Text'}</span>
+            <CaretDown className="caret" />
+        </div>
+    );
+}, style);
+
+export interface DropDownListProps {
+    open: boolean;
+    items?: DropDownItem[];
```

Comment:
Decorating the DOM elements with `data-` attributes allows having a single onClick handler on the container handling all clicks for its children. It requires being able to know the unique string representation of the item.

In `<TreeView />`, the current code avoids converting an item to a string by creating a **unique** `onClick` **function** (bound to current item) **for each** `<TreeItem />`. This is costly if the tree has many items.

@idoros suggested a possible pattern regarding a `dataResolver` and a `dataRenderer` that could possibly address this.

Not sure what's the ideal pattern in this case. need to think further about it.